compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/FileSystemRepository.java

```diff
@@ -172,9 +173,54 @@ public FileSystemRepository(Environment environment, Path cacheDirectory, String
         } catch (IOException e) {
             throw new RuntimeException("Error while accessing Distribution cache: " + e.getMessage());
         }
+
+        List<Path> incompatibleVersions = new ArrayList<>();
+
+        if (!versions.isEmpty()) {
+            for (Path ver : versions) {
+                Path pkgJsonPath = getPackagePath(org.value(), name.value(), ver.getFileName().toString())
+                        .resolve(ProjectConstants.PACKAGE_JSON);
+
+                if (Files.exists(pkgJsonPath)) {
+                    String packageVer = BalaFiles.readPkgJson(pkgJsonPath).getBallerinaVersion();
+                    String packVer = RepoUtils.getBallerinaShortVersion();
+
+                    if (!packageVer.equals(packVer)) {
```

Comment:
We have done this check inside the isCompatible as well? we can remove the from one place.

Reply:
Removed extra checks